packages/lmi/src/lmi/types.py

```diff
@@ -79,6 +79,9 @@ class LLMResult(BaseModel):
     logprob: float | None = Field(
         default=None, description="Sum of logprobs in the completion."
     )
+    top_logprobs: list[list[tuple[str, float]]] = Field(
+        default_factory=list, description="Top logprobs for each position in the completion."
+    )
```

Comment:
I would rather see `list[list[tuple[str, float]]] | None` with `default=None` to avoid injecting information such as empty list